src/topmodel.c

```diff
@@ -508,9 +508,189 @@ if(yes_print_output==TRUE)
 return;
 }
 
+
+/*********************************
+ * Define functions to be used in init() and in Set_value in bmi_topmodel.c
+ * to enable calibratable parameters to be updated
+ * **********************************/
+
+
+// FUNCTION TO CONVERT DISTANCE/AREA FORM TO TIME DELAY HISTOGRAM ORDINATES
```

Comment:
Definitely reccomend documenting function paramers inline.  using [doxygen style comments ](https://www.doxygen.nl/manual/docblocks.html) can also help auto generate human readable docs in the future.  Consider a comment like this
```C
/**
* Function to convert Distance/Area from T0 time delay historgram ordinates.
*
* <additional details, also known as long description, here e.g.>
* This function converts paramters to m/time step DT with exception of XK0
* which must stay in m/h.  
* Q0 is already in m/time step
* T0 is input as Ln(To)
* 
* @param[in] dist_from_outlet pointer to array of length num_channels containing WHAT
* @param[in] num_channels the number of channels, corresponds to the length of dist_from_outlet and tch arrays
* @param[in] chv <is what???>
* @param[in] rv <is what???>
* @param[in] dt delta time, or time step, used to convert chv and rv
* @param[out] tch <what is tch, generated by this function>
*/
```
I'll also note that I don't see an `XK0` or `T0` parameter anywhere in this function, so perhaps that comment needs to be dropped???